querybook/webapp/components/DataDocCell/DataDocCell.tsx

```diff
@@ -80,6 +92,60 @@ export const DataDocCell: React.FunctionComponent<IDataDocCellProps> =
                 fullScreenCellIndex,
             } = useContext(DataDocContext);
 
+            let queryCells = useQueryCells(docId);
+            const queryTitle = queryCells[index].meta.title;
+            const title =
+                queryTitle == null || queryTitle === ''
+                    ? 'Query #' + (index + 1).toString()
+                    : queryTitle;
+            queryCells = queryCells.slice(index);
+
+            const latestQueryExecutions = useMakeSelector(
+                makeLatestQueryExecutionsSelector,
+                queryCells.map((c) => c.id) ?? []
+            );
+            const hasQueryRunning = useMemo(
+                () => latestQueryExecutions.some((q) => q.status < 3),
+                [latestQueryExecutions]
+            );
+            const notification = useRef(true);
+
+            const onRunAll = useCallback(() => {
+                sendConfirm({
+                    header: 'Run Cells From ' + title,
+                    message: (
+                        <DataDocRunAllButtonConfirm
```

Comment:
i felt for things like hasQueryRunning, queryCells just be within the `DataDocRunAllButtonConfirm` since none of these computation are actually needed until the message is shown.

@jczhong84 what is your thought on that

Reply:
agreed. `DataDocRunAllButtonConfirm` could be reused here with additional props passed over.